plugins/kubernetes-backend/src/service/KubernetesFetcher.ts

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  V1ConfigMap,
+  V1Deployment,
+  V1Pod,
+  V1ReplicaSet,
+  V1Secret,
+} from '@kubernetes/client-node';
+import { ClusterDetails } from '../cluster-locator/types';
+import { KubernetesClientProvider } from './KubernetesClientProvider';
+import { V1Service } from '@kubernetes/client-node/dist/gen/model/v1Service';
+import { Clients } from './types';
+import { Logger } from 'winston';
+
+export interface KubernetesFetcher {
+  fetchServicesByServiceId(
```

Comment:
What do you think about making this more of a `fetchResources({thingsToFetch: Set<'services' | 'pods'>}): Promise<Array<{type: ... , resource: ...}>>` thing? Kinda wondering how CRDs would be supported with this interface, and that question extends all the way to the k8s client api :grin:

Reply:
I think that sounds great! Then there could be a bit of config work later to configure what objects to fetch for each use case.